File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/entities/redis/OidcAuthorizationCodeGrantAuthorization.java
package com.sv.youapp.component.authorization.entities.redis;

import lombok.Getter;
import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest;

import java.security.Principal;
import java.time.Instant;

@Getter
public class OidcAuthorizationCodeGrantAuthorization extends OAuth2AuthorizationCodeGrantAuthorization {

	private final IdToken idToken;

	public OidcAuthorizationCodeGrantAuthorization(
		String id,
		String registeredClientId,
		String principalName,
		String scopes,
		AccessToken accessToken,
		RefreshToken refreshToken,
		Principal principal,
		OAuth2AuthorizationRequest authorizationRequest,
		AuthorizationCode authorizationCode,
		String state,
		IdToken idToken
	) {
		super(id, registeredClientId, principalName, scopes, accessToken, refreshToken, principal, authorizationRequest, authorizationCode, state);
		this.idToken = idToken;
	}

	@Getter
	public static final class IdToken extends AbstractToken {
		private final AbstractGrantAuthorization.ClaimsHolder claims;

		public IdToken(
			String tokenValue,
			Instant issuedAt,
			Instant expiresAt,
			boolean invalidated,
			AbstractGrantAuthorization.ClaimsHolder claims
		) {
			super(tokenValue, issuedAt, expiresAt, invalidated);
			this.claims = claims;
		}

	}
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/service/authorization/entities/redis/OidcAuthorizationCodeGrantAuthorization.kt
package com.sv.youapp.service.authorization.entities.redis

import org.springframework.security.oauth2.core.endpoint.OAuth2AuthorizationRequest
import java.security.Principal
import java.time.Instant

class OidcAuthorizationCodeGrantAuthorization(
    id: String,
    registeredClientId: String,
    principalName: String,
    scopes: String,
    accessToken: AccessToken?,
    refreshToken: RefreshToken?,
    principal: Principal?,
    authorizationRequest: OAuth2AuthorizationRequest?,
    authorizationCode: AuthorizationCode?,
    state: String?,
    val idToken: IdToken?,
) : OAuth2AuthorizationCodeGrantAuthorization(id, registeredClientId, principalName, scopes, accessToken, refreshToken, principal, authorizationRequest, authorizationCode, state) {
    class IdToken(
        tokenValue: String?,
        issuedAt: Instant?,
        expiresAt: Instant?,
        invalidated: Boolean,
        val claims: ClaimsHolder?,
    ) : AbstractToken(tokenValue, issuedAt, expiresAt, invalidated)
}
